extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/S3StorageConfig.java

```diff
@@ -36,17 +36,34 @@
   @JsonProperty("sse")
   private final ServerSideEncryption serverSideEncryption;
 
+  /**
+   * S3 transfer config.
+   *
+   * @see S3StorageDruidModule#configure
+   */
+  @JsonProperty("transfer")
```

Comment:
I see we use this on other properties, but I'm not sure why those are needed, the property is already defined in the constructor and as a getter.

Reply:
> I see we use this on other properties, but I'm not sure why those are needed, the property is already defined in the constructor and as a getter.

I turns out the annotation on the field is required. I created a test version without the annotation, and got the following errors:
```
{
  "@timestamp": "2020-08-26T03:06:15.352Z",
  "log.level": "ERROR",
  "message": "Error when starting up.  Failing.",
  "service.name": "druid/middleManager",
  "event.dataset": "druid/middleManager.log",
  "process.thread.name": "main",
  "log.logger": "org.apache.druid.cli.CliMiddleManager",
  "error.type": "com.google.inject.ProvisionException",
  "error.message": "Unable to provision, see the following errors:

1) JsonConfigurator requires Jackson-annotated Config objects to have field annotations. class org.apache.druid.storage.s3.S3StorageConfig doesn't
  at org.apache.druid.guice.JsonConfigProvider.bind(JsonConfigProvider.java:151) (via modules: com.google.inject.util.Modules$OverrideModule -> org.apache.druid.storage.s3.S3StorageDruidModule)
  at org.apache.druid.guice.JsonConfigProvider.bind(JsonConfigProvider.java:151) (via modules: com.google.inject.util.Modules$OverrideModule -> org.apache.druid.storage.s3.S3StorageDruidModule)
  while locating com.google.common.base.Supplier<org.apache.druid.storage.s3.S3StorageConfig>
  at org.apache.druid.guice.JsonConfigProvider.bind(JsonConfigProvider.java:152) (via modules: com.google.inject.util.Modules$OverrideModule -> org.apache.druid.storage.s3.S3StorageDruidModule)
  while locating org.apache.druid.storage.s3.S3StorageConfig
    for the 5th parameter of org.apache.druid.storage.s3.S3StorageDruidModule.getServerSideEncryptingAmazonS3Builder(S3StorageDruidModule.java:183)
  while locating org.apache.druid.storage.s3.ServerSideEncryptingAmazonS3$Builder
    for the 1st parameter of org.apache.druid.storage.s3.S3StorageDruidModule.getAmazonS3Client(S3StorageDruidModule.java:218)
  at org.apache.druid.storage.s3.S3StorageDruidModule.getAmazonS3Client(S3StorageDruidModule.java:218) (via modules: com.google.inject.util.Modules$OverrideModule -> org.apache.druid.storage.s3.S3StorageDruidModule)
  while locating org.apache.druid.storage.s3.ServerSideEncryptingAmazonS3
    for the 1st parameter of org.apache.druid.storage.s3.S3TaskLogs.<init>(S3TaskLogs.java:60)
  at org.apache.druid.storage.s3.S3StorageDruidModule.configure(S3StorageDruidModule.java:168) (via modules: com.google.inject.util.Modules$OverrideModule -> org.apache.druid.storage.s3.S3StorageDruidModule)
  while locating org.apache.druid.storage.s3.S3TaskLogs
  while locating org.apache.druid.tasklogs.TaskLogs annotated with @com.google.inject.multibindings.Element(setName=,uniqueId=129, type=MAPBINDER, keyType=java.lang.String)
  at org.apache.druid.guice.PolyBind.createChoice(PolyBind.java:71) (via modules: com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> org.apache.druid.guice.IndexingServiceTaskLogsModule)
  while locating org.apache.druid.tasklogs.TaskLogs
  while locating org.apache.druid.tasklogs.TaskLogPusher
    for the 5th parameter of org.apache.druid.indexing.overlord.ForkingTaskRunner.<init>(ForkingTaskRunner.java:116)
  at org.apache.druid.cli.CliMiddleManager$1.configure(CliMiddleManager.java:113) (via modules: com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> org.apache.druid.cli.CliMiddleManager$1)
  while locating org.apache.druid.indexing.overlord.ForkingTaskRunner
  while locating org.apache.druid.indexing.overlord.TaskRunner
    for the 2nd parameter of org.apache.druid.indexing.worker.WorkerTaskMonitor.<init>(WorkerTaskMonitor.java:72)
  at org.apache.druid.cli.CliMiddleManager.bindWorkerManagementClasses(CliMiddleManager.java:197) (via modules: com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> org.apache.druid.cli.CliMiddleManager$1)
  while locating org.apache.druid.indexing.worker.WorkerTaskMonitor

1 error",
  "error.stack_trace": "com.google.inject.ProvisionException: Unable to provision, see the following errors:

1) JsonConfigurator requires Jackson-annotated Config objects to have field annotations. class org.apache.druid.storage.s3.S3StorageConfig doesn't
  at org.apache.druid.guice.JsonConfigProvider.bind(JsonConfigProvider.java:151) (via modules: com.google.inject.util.Modules$OverrideModule -> org.apache.druid.storage.s3.S3StorageDruidModule)
  at org.apache.druid.guice.JsonConfigProvider.bind(JsonConfigProvider.java:151) (via modules: com.google.inject.util.Modules$OverrideModule -> org.apache.druid.storage.s3.S3StorageDruidModule)
  while locating com.google.common.base.Supplier<org.apache.druid.storage.s3.S3StorageConfig>
  at org.apache.druid.guice.JsonConfigProvider.bind(JsonConfigProvider.java:152) (via modules: com.google.inject.util.Modules$OverrideModule -> org.apache.druid.storage.s3.S3StorageDruidModule)
  while locating org.apache.druid.storage.s3.S3StorageConfig
    for the 5th parameter of org.apache.druid.storage.s3.S3StorageDruidModule.getServerSideEncryptingAmazonS3Builder(S3StorageDruidModule.java:183)
  while locating org.apache.druid.storage.s3.ServerSideEncryptingAmazonS3$Builder
    for the 1st parameter of org.apache.druid.storage.s3.S3StorageDruidModule.getAmazonS3Client(S3StorageDruidModule.java:218)
  at org.apache.druid.storage.s3.S3StorageDruidModule.getAmazonS3Client(S3StorageDruidModule.java:218) (via modules: com.google.inject.util.Modules$OverrideModule -> org.apache.druid.storage.s3.S3StorageDruidModule)
  while locating org.apache.druid.storage.s3.ServerSideEncryptingAmazonS3
    for the 1st parameter of org.apache.druid.storage.s3.S3TaskLogs.<init>(S3TaskLogs.java:60)
  at org.apache.druid.storage.s3.S3StorageDruidModule.configure(S3StorageDruidModule.java:168) (via modules: com.google.inject.util.Modules$OverrideModule -> org.apache.druid.storage.s3.S3StorageDruidModule)
  while locating org.apache.druid.storage.s3.S3TaskLogs
  while locating org.apache.druid.tasklogs.TaskLogs annotated with @com.google.inject.multibindings.Element(setName=,uniqueId=129, type=MAPBINDER, keyType=java.lang.String)
  at org.apache.druid.guice.PolyBind.createChoice(PolyBind.java:71) (via modules: com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> org.apache.druid.guice.IndexingServiceTaskLogsModule)
  while locating org.apache.druid.tasklogs.TaskLogs
  while locating org.apache.druid.tasklogs.TaskLogPusher
    for the 5th parameter of org.apache.druid.indexing.overlord.ForkingTaskRunner.<init>(ForkingTaskRunner.java:116)
  at org.apache.druid.cli.CliMiddleManager$1.configure(CliMiddleManager.java:113) (via modules: com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> org.apache.druid.cli.CliMiddleManager$1)
  while locating org.apache.druid.indexing.overlord.ForkingTaskRunner
  while locating org.apache.druid.indexing.overlord.TaskRunner
    for the 2nd parameter of org.apache.druid.indexing.worker.WorkerTaskMonitor.<init>(WorkerTaskMonitor.java:72)
  at org.apache.druid.cli.CliMiddleManager.bindWorkerManagementClasses(CliMiddleManager.java:197) (via modules: com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> org.apache.druid.cli.CliMiddleManager$1)
  while locating org.apache.druid.indexing.worker.WorkerTaskMonitor

1 error
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1028)
	at com.google.inject.internal.InjectorImpl.getInstance(InjectorImpl.java:1050)
	at org.apache.druid.guice.LifecycleModule$2.start(LifecycleModule.java:141)
	at org.apache.druid.cli.GuiceRunnable.initLifecycle(GuiceRunnable.java:115)
	at org.apache.druid.cli.ServerRunnable.run(ServerRunnable.java:63)
	at org.apache.druid.cli.Main.main(Main.java:113)
"
}
```